### DISCUSSION TITLE
The new version of magick, gif to webp is slower

### DISCUSSION THREAD

**Discussion by nanfeng1999** [ImageMagick/ImageMagick#6661] [Help] [answered]
### ImageMagick version

7.1.1-11

### Operating system, version and so on

CentOS release 6.3 (Final)

### Description

hello everyone! I would like to know why the new version of magick has a slower speed when converting gif to webp format. Is there any solution?

>  **bash-4.1$ ./magick --version**
Version: ImageMagick 7.0.10-47 Q16 x86_64 2022-01-12 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI 
Delegates (built-in): bzlib freetype heic jng jp2 jpeg lcms lzma png tiff webp zlib
**bash-4.1$ ./magick-new --version**
Version: ImageMagick 7.1.1-11 Q16-HDRI x86_64 f04a7eb33:20230528 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Zero-configuration 
Delegates (built-in): bzlib freetype heic jng jp2 jpeg lcms lzma png tiff webp zlib
Compiler: gcc (12.1)
**bash-4.1$ time ./magick convert origin.gif origin.webp**
real    0m7.807s
user    0m7.151s
sys     0m0.655s
**bash-4.1$ time ./magick-new convert origin.gif origin.webp**
real    0m15.283s
user    0m14.507s
sys     0m0.776s

**Comment by fmw42** [comment]
On IM 7, use magick, not magick convert.

**Comment by nanfeng1999** [comment]
**bash-4.1$ time ./magick origin.gif origin.webp**
real    0m8.047s
user    0m7.201s
sys     0m0.844s
**bash-4.1$ time ./magick-new origin.gif origin.webp**
real    0m15.233s
user    0m14.508s
sys     0m0.725s

The two commands are equivalent @fmw42

**Comment by urban-warrior** [comment]
Can you post your `origin.gif` image here or a link to it so we can download it and reproduce the issue?

**Comment by nanfeng1999** [comment]
Of course, here is the link: `https://img.ithome.com/newsuploadfiles/2023/8/8d151558-37ea-4418-9d8e-11e2c7e7fce6.gif`
@urban-warrior

**Comment by snibgo** [comment]
I can't reproduce the problem. An old version of IM (7.1.0-54) elapsed time seems slower than a recent version (7.1.1-15). This is on Windows 11 with SSD.

```
TimeC c:\im\ImageMagick-7.1.0-54-Q16-HDRI\magick C:\web\im\scrollWatch.gif x.webp
Seconds: 7.16

D:\temp>timec %IMG71054%magick C:\web\im\scrollWatch.gif x.webp

TimeC c:\im\ImageMagick-7.1.0-54-Q16-HDRI\magick C:\web\im\scrollWatch.gif x.webp
Seconds: 7.17


D:\temp>timec %IMG71115%magick C:\web\im\scrollWatch.gif x.webp

TimeC C:\im\ImageMagick-7.1.1-15-Q16-HDRI\magick C:\web\im\scrollWatch.gif x.webp
Seconds: 4.87

D:\temp>timec %IMG71115%magick C:\web\im\scrollWatch.gif x.webp

TimeC C:\im\ImageMagick-7.1.1-15-Q16-HDRI\magick C:\web\im\scrollWatch.gif x.webp
Seconds: 5.04
```

**Comment by urban-warrior** [comment]
We no longer automatically coalesce the image sequence for WebP.  As a result, compression may take longer and result in a larger WebP image file.  To replicate the previous behavior, use

```
time ./magick origin.gif -coalesce origin.webp
```

**Comment by nanfeng1999** [comment]
Can you tell me the version of your libwebp? @snibgo

**Comment by nanfeng1999** [comment]
here is my configure parameter. May it reduce performance？

./configure --prefix=${prefix}    --enable-static  --disable-shared --disable-dependency-tracking  --disable-silent-rules  --disable-opencl --disable-openmp --enable-zero-configuration

**Comment by snibgo** [comment]
> Can you tell me the version of your libwebp? @snibgo

```
D:\temp>%IMG71054%magick -list format |grep -i webp
     WEBP* rw+   WebP Image Format (libwebp 1.2.2 [020F])

D:\temp>%IMG71115%magick -list format |grep -i webp
     WEBP* rw+   WebP Image Format (libwebp 1.3.1 [020F])
```

**Comment by nanfeng1999** [comment]
> here is my configure parameter. May it reduce performance？
> 
> ./configure --prefix=${prefix} --enable-static --disable-shared --disable-dependency-tracking --disable-silent-rules --disable-opencl --disable-openmp --enable-zero-configuration

Can you take a look at this question @urban-warrior

**Comment by urban-warrior** [comment]
Your configure parameters do not affect the performance of your command-line.  Add `-define webp:thread-level=1` to your command-line to increase performance by 5-10%.

Your image has 132 frames so its not surprising it takes several seconds to complete the conversion with compression.

**Comment by nanfeng1999** [comment, marked as answer]
Thank you very much for your suggestion! What's strange now is that the old version took normal time, while the new version is not working properly. We are still investigating this reason @urban-warrior

**Comment by nanfeng1999** [comment]
`./magick origin.gif origin.webp`
May I ask if this command generates a dynamic webp or multiple static webps according to the current normal logic @urban-warrior

**Comment by urban-warrior** [comment]
For image sequences, we call the `libwebp` method WebPAnimEncoderNew() followed by WriteSingleWEBPPicture().  Take a look at the Webp delegate library documentation to clarify what kind of image results you can expect from these methods.
